x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/UnresolvedRelation.java

```diff
@@ -30,21 +30,24 @@ public class UnresolvedRelation extends LeafPlan implements Unresolvable {
     private final List<Attribute> metadataFields;
     private final IndexMode indexMode;
     private final String unresolvedMsg;
+    private final String commandName;
```

Comment:
Could you think of another solution instead of this? Maybe an interface (Measureable or something) that returns the command name or something that doesn't change the properties of this class? I fear that we make these changes here that are not related to logical plans to cover functionality that should just look at or inspect logical plans. An UnresolvedRelation shouldn't be concerned with metrics imho.

Reply:
I thought about this for some time, but I still can't find a reasonable solution (I agree with you in principle, I don't like having the `commandName` there).

The problem is that we collect statistics at pre-analysis time; at that stage we only have the logical plan, and the plans of very different queries can be identical, 

Eg.

```
FROM idx | ...
```
```
METRICS idx | ...
```

will both translate to `UnresolvedRelation`, practically impossible to distinguish.

As a side note, we have the exact same problem with the legacy stats: a query with `METRICS idx |` will be tracked as a normal `FROM idx`.

The only solution I can see is to move the calculation of the stats into the parser. 
Eg. inside `LogialPlanBuilder.visitMetricsCommand()` we know that an `UnresolvedRelation` comes from a `METRICS` command, but right after that we lose this information completely (unless we store it in the plan itself).

The problem with this approach is that we'll have to scatter the logic in all the `visitXYZCommand()` methods, that is not very convenient and hard to maintain.

An alternative could be to just ignore this corner case for now and accept that `METRICS idx` is logged as a `FROM idx`. Functionally they are the same, but I really don't like the idea of losing the user intention.

Any suggestions are welcome

Reply:
Add a comment here mentioning what `commandName` represents and its usage. It may be surprising to other people looking at the code. And/or a more meaningful name.... `telemetryMetricName`?